sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java

```diff
@@ -1079,6 +1539,166 @@ public TableTransactionResult submitTransaction(List<TableTransactionAction> tra
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<TableTransactionResult> submitTransactionWithResponse(List<TableTransactionAction> transactionActions, Duration timeout, Context context) {
-        return blockWithOptionalTimeout(client.submitTransactionWithResponse(transactionActions, context), timeout);
+        OptionalLong timeoutInMillis = TableUtils.setTimeout(timeout);
+        Context contextValue = TableUtils.setContext(context, true);
+
+        if (transactionActions.isEmpty()) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("A transaction must contain at least one operation."));
+        }
+
+        final List<TransactionalBatchAction> operations = new ArrayList<>();
+
+        for (TableTransactionAction transactionAction : transactionActions) {
+            switch (transactionAction.getActionType()) {
+                case CREATE:
+                    operations.add(new TransactionalBatchAction.CreateEntity(transactionAction.getEntity()));
+
+                    break;
+                case UPSERT_MERGE:
+                    operations.add(new TransactionalBatchAction.UpsertEntity(transactionAction.getEntity(),
+                        TableEntityUpdateMode.MERGE));
+
+                    break;
+                case UPSERT_REPLACE:
+                    operations.add(new TransactionalBatchAction.UpsertEntity(transactionAction.getEntity(),
+                        TableEntityUpdateMode.REPLACE));
+
+                    break;
+                case UPDATE_MERGE:
+                    operations.add(new TransactionalBatchAction.UpdateEntity(transactionAction.getEntity(),
+                        TableEntityUpdateMode.MERGE, transactionAction.getIfUnchanged()));
+
+                    break;
+                case UPDATE_REPLACE:
+                    operations.add(new TransactionalBatchAction.UpdateEntity(transactionAction.getEntity(),
+                        TableEntityUpdateMode.REPLACE, transactionAction.getIfUnchanged()));
+
+                    break;
+                case DELETE:
+                    operations.add(
+                        new TransactionalBatchAction.DeleteEntity(transactionAction.getEntity(),
+                            transactionAction.getIfUnchanged()));
+
+                    break;
+                default:
+                    break;
+            }
```

Comment:
This code can be shared with async client, consider moving to util.

Reply:
Thank you, I'll mark this for a refactoring PR